default/python/AI/AIstate.py

```
@@ -986,3 +991,26 @@ def log_war_declaration(self, initiating_empire_id, recipient_empire_id):
 
     def get_standard_enemy(self):
         return CombatRatingsAI.ShipCombatStats(stats=self.__empire_standard_enemy)
+
+    def surrender(self):
+        chat_human("Well played. I surrender.")
+        chat_human("NOTE: The AI will accept any peace or alliance proposal at this point.")
+        self.surrendered = True
+        my_empire_id = fo.empireID()
+        for empire_id in fo.allEmpireIDs():
+            if empire_id == my_empire_id or fo.playerIsAI(empire_id):
+                continue
+            try:
+                msg = fo.diplomaticMessage(my_empire_id, empire_id,
+                                           fo.diplomaticMessageType.peaceProposal)
+                debug("Sending peace proposal")
+                fo.sendDiplomaticMessage(msg)
+            except:
```

Bare except.   May be let it fail? It will be handled later.